### PR TITLE
Run INa IV protocols

### PR DESCRIPTION
**Pull request by MichaelClerx**
See #99 

Still needs

- [x] unit conversion rules #156 
- [x] clamping optional values to initial variables #177

### CODE REVIEW

**Comment by MichaelClerx** [comment]
Why is this called "optional variables"? Doesn't include any optional inputs/output, but instead tests `var` declarations and unit conversion rules.

**Comment by jonc125** [comment]
One test protocol has things like `optional oxmeta:extracellular_sodium_concentration`, and another things like
```
optional oxmeta:temperature default 310 :: K
optional oxmeta:sodium_reversal_potential default (R*oxmeta:temperature/F) * MathML:ln(oxmeta:extracellular_sodium_concentration / oxmeta:cytosolic_sodium_concentration)
```

It's just those aren't the only new features in those protocols. You could create new smaller test cases for individual features if you like.

**Comment by MichaelClerx** [comment]
@jonc125 I think this failure has nothing to do with this PR but must come from here:

```
    def __init__(self):
        """Initialise the parser."""
        # We just store the original stack limit here, so we can increase
        # it for the lifetime of this object if needed for parsing, on the
        # basis that if one expression needs to, several are likely to.
        self._stack_depth_factor = 1
        self._original_stack_limit = sys.getrecursionlimit()
        self.increase_stack_depth_limit()

    def __del__(self):
        """Reset the stack limit if it changed."""
        sys.setrecursionlimit(self._original_stack_limit)

    def increase_stack_depth_limit(self, step=0.5):
        """Increase the limit by the given factor of the original."""
        self._stack_depth_factor += 0.5
        new_limit = int(
            self._stack_depth_factor * self._original_stack_limit)
        print('Increasing recursion limit to', new_limit,
              file=sys.stderr)
        sys.setrecursionlimit(new_limit)
```

I guess every CSP created in the same process increases the stack limit further? Seems a bad idea...

**Comment by MichaelClerx** [comment]
Got rid of the CSP thing, but 2/3 tests still fail. No idea why or what those error messages mean.
Can you take this one up, @jonc125 ?

**Comment by MichaelClerx** [comment]
Remaining issue seems to be https://github.com/ModellingWebLab/cellmlmanip/issues/278

File: fc/parsing/actions.py [hunk 1]
```diff
@@ -1914,14 +1914,14 @@ def _handle_clamping(self):
                         raise ProtocolError(f'Clamp-to-initial-value set for unknown variable {pvar.long_name}.')
 
                 # Evaluate and update equation
-                defn = self.model.get_definition(var)
-                if defn.lhs.is_Derivative:
+                if self.model.is_state(var):
                     value = var.initial_value
                 else:
-                    value = defn.rhs.evalf()  # TODO: What if there are variable references in the RHS?
+                    value = self.model.get_value(var)
```
Comment (jonc125) [inline fc/parsing/actions.py:1920]: We could just use `value = self.model.get_value(var)` without the check now!

File: test/test_iv_curves.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    proto.run()
+
+    # Check the key outputs haven't changed.
+    peaks = proto.output_env.look_up('normalised_peak_currents').array
+    expected = [1.0, 0.999998, 0.999981, 0.999906, 0.999564, 0.999062, 0.99795, 0.995344, 0.988466, 0.966859, 0.887918,
+                0.624599, 0.192671, 0.0186169, 0.00193235, 0.00116954, 0.00114086, 0.00113976, 0.00113976]
+    assert len(peaks) == len(expected)
+    ipeaks = iter(peaks)
+    for value in expected:
+        assert pytest.approx(value, 1e-3) == next(ipeaks)
```
Comment (jonc125) [inline test/test_iv_curves.py:24]:
```suggestion
        assert next(ipeaks) == pytest.approx(value, 1e-3)
```

File: test/test_iv_curves.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (jonc125) [inline test/test_iv_curves.py:42]:
```suggestion
        assert next(ipeaks) == pytest.approx(value, 1e-3)
```

File: test/test_iv_curves.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    proto.run()
+
+    # Check the key outputs haven't changed.
+    peaks = proto.output_env.look_up('normalised_peak_currents').array
+    expected = [0.893527, 0.946238, 0.980761, 0.999288, 1, 0.98876, 0.961362, 0.903678, 0.79484, 0.622032, 0.410937,
+                0.224468, 0.104744, 0.044427, 0.018137, 0.00741457, 0.00312374, 0.000728363, 0.000343776]
+    assert len(peaks) == len(expected)
+    ipeaks = iter(peaks)
+    for value in expected:
+        assert pytest.approx(value, 1e-3) == next(ipeaks)
```
Comment (jonc125) [inline test/test_iv_curves.py:59]:
```suggestion
        assert next(ipeaks) == pytest.approx(value, 1e-3)
```

**Pull request merged**